### ISSUE TITLE
smux: port muxcool's deliver_now/Parked flow control (session-wide head-of-line blocking) + NOP keepalive

### ISSUE THREAD

**Issue by madeye** [madeye/mihomo-rust#425]
Deferred from the #414 review (finding #1). `STREAM_QUEUE` was raised 2 → 32,
which moves the threshold but leaves the mechanism unchanged: the smux reader
task still awaits the bounded per-stream channel in
`crates/meow-proxy/src/mux/smux.rs`:

```rust
result = tx.send(chunk) => { let _ = result; }
```

A consumer that stops reading blocks the reader task, which halts frame dispatch
for **every** stream on that physical connection — one paused download stalls
every other tab sharing the session. The 4 MiB `receive_budget` semaphore already
bounds memory, so the bounded channel buys no safety here, only stalls.

`mux/muxcool.rs` (#418) implements the target shape and is the reference:
`deliver_now` / `Parked` / the `space` `Notify` park at most one frame and pause
the connection read (session-wide flow control, TCP window semantics) without the
driver ever blocking on a stream queue — so a consumer stalled mid-write cannot
deadlock writers sharing the session.

**Action:** port that pattern to smux. The `STREAM_QUEUE = 32` comment in
smux.rs already names this as the intended follow-up; this issue tracks it so it
does not live only in a code comment.

**Also deferred from the same review (#7):** smux has no `CMD_NOP` keepalive.
xtaci and sing-mux both send them; without one, a half-dead session is only
detected on an IO failure or after the 60 s idle eviction, so a stream opened
onto a silently-dead session pays the full `open_stream_flags` retry.
`mux/muxcool.rs`'s `keepalive_loop` is the in-tree precedent.

**Comment by madeye** [comment]
@aimkiray — owner, please. Deferred by agreement in the #414 review (findings #1 and #7). Your `deliver_now` / `Parked` / `space` design in `mux/muxcool.rs` is the reference implementation; this is the port to smux, plus the `CMD_NOP` keepalive.

**Comment by aimkiray** [comment]
Taking this — the smux scaffold in #414 was mine, so the flow-control follow-up lands naturally with me.

Plan: port muxcool's `deliver_now` / `Parked` / `space` Notify delivery to `smux.rs`, so a full per-stream queue parks at most one frame and pauses the connection read (session-wide flow control) instead of blocking the shared reader task on the bounded channel.

One factual correction on the keepalive half: sing-mux does **not** send NOPs. Both the SagerNet and metacubex forks set `config.KeepAliveDisabled = true` in `smuxConfig()` (session.go), and neither adds an app-level keepalive. xtaci's upstream smux does send them (10s default), but the mihomo stack deliberately runs with keepalive disabled — our `smux.rs` module doc already records this (`DefaultConfig + KeepAliveDisabled`). I'll evaluate whether adding a NOP keepalive is worth deviating from mihomo's behavior as part of the work, and scope the PR accordingly.

**Comment by aimkiray** [comment]
The flow-control half (the Action in the issue body) landed in #491. One scoping note so the headline symptom isn't misread as fixed:

**"One paused download stalls every other tab sharing the session" remains — by design.** The muxcool `deliver_now`/`Parked` shape that #491 ports pauses the connection read while a chunk is parked; that pause *is* the session-wide flow control (TCP window semantics) the issue's own reference description names. A stalled consumer therefore still backpressures every stream on the session, both before and after the port. Avoiding that would mean keeping the read going and buffering toward the 4 MiB `receive_budget` instead of parking — a deliberate deviation from muxcool/mihomo behavior that deserves its own discussion if anyone wants it.

What the port actually changes is the mechanism: the reader never suspends inside one stream's channel wait queue. In muxcool that matters because a single driver blocked on a stream queue starves the outbound side (deadlock for a consumer stalled mid-write); in smux the writer is already a separate task, so the port is behavior-preserving there — it buys structural consistency with the reference, explicit park semantics pinned by tests, and deadlock-safety if the session ever moves to a unified driver.

Still open on this issue: the NOP-keepalive half (with the sing-mux `KeepAliveDisabled = true` caveat from the discussion).
